### PR TITLE
- added encrypted shared preferences.

### PR DESCRIPTION
**Pull request by amit-metri**
- added encrypted shared preferences.
- older preferences are ignored.

### CODE REVIEW

File: app/src/main/java/ca/bc/gov/bchealth/datasource/DataStoreRepo.kt [hunk 1]
```diff
@@ -1,69 +1,73 @@
...
 
-    val isRecentFormData: Flow<String> = context.dataStore.data.map { preference ->
-        preference[RECENT_FORM_DATA] ?: ""
+    val isRecentFormData: Flow<String> = flow {
+        emit(encryptedPreferences.getString(RECENT_FORM_DATA, "") ?: "")
     }
 
-    suspend fun setRecentFormData(formData: String) = context.dataStore.edit { preference ->
-        preference[RECENT_FORM_DATA] = formData
+    fun setRecentFormData(formData: String) {
```
Comment (pinakinkansara) [inline app/src/main/java/ca/bc/gov/bchealth/datasource/DataStoreRepo.kt:68]: mark this as suspend function

File: app/src/main/java/ca/bc/gov/bchealth/datasource/DataStoreRepo.kt [hunk 2]
```diff
@@ -1,69 +1,73 @@
...
-        preference[NEW_FEATURE] ?: false
+    val isNewFeatureShown: Flow<Boolean> = flow {
+        emit(
+            encryptedPreferences.getBoolean(NEW_FEATURE, false)
+        )
     }
 
-    suspend fun setNewFeatureShown(shown: Boolean = true) = context.dataStore.edit { preference ->
-        preference[NEW_FEATURE] = shown
+    fun setNewFeatureShown(shown: Boolean = true) {
```
Comment (pinakinkansara) [inline app/src/main/java/ca/bc/gov/bchealth/datasource/DataStoreRepo.kt:58]: mark this as suspend function

File: app/src/main/java/ca/bc/gov/bchealth/datasource/DataStoreRepo.kt [hunk 3]
```diff
@@ -1,69 +1,73 @@
...
 
-    val isAnalyticsEnabled: Flow<Boolean> = context.dataStore.data.map { preference ->
-        preference[IS_ANALYTICS_ENABLED] ?: true
+    val isAnalyticsEnabled: Flow<Boolean> = flow {
+        emit(encryptedPreferences.getBoolean(IS_ANALYTICS_ENABLED, true))
     }
 
-    suspend fun trackAnalytics(shown: Boolean = true) = context.dataStore.edit { preference ->
-        preference[IS_ANALYTICS_ENABLED] = shown
+    fun trackAnalytics(shown: Boolean = true) {
```
Comment (pinakinkansara) [inline app/src/main/java/ca/bc/gov/bchealth/datasource/DataStoreRepo.kt:46]: mark this as suspend function

File: app/src/main/java/ca/bc/gov/bchealth/datasource/DataStoreRepo.kt [hunk 4]
```diff
@@ -1,69 +1,73 @@
...
 
-    val isOnBoardingShown: Flow<Boolean> = context.dataStore.data.map { preference ->
-        preference[ON_BOARDING_SHOWN] ?: false
+    val isOnBoardingShown: Flow<Boolean> = flow {
+        emit(encryptedPreferences.getBoolean(ON_BOARDING_SHOWN, false))
     }
 
-    suspend fun setOnBoardingShown(shown: Boolean = true) = context.dataStore.edit { preference ->
-        preference[ON_BOARDING_SHOWN] = shown
+    fun setOnBoardingShown(shown: Boolean = true) {
```
Comment (pinakinkansara) [inline app/src/main/java/ca/bc/gov/bchealth/datasource/DataStoreRepo.kt:36]: mark this as suspend function

File: app/src/main/java/ca/bc/gov/bchealth/di/DataStoreModule.kt [hunk 5]
```diff
@@ -20,5 +24,25 @@ class DataStoreModule {
...
-    fun providesDataStore(@ApplicationContext context: Context) = DataStoreRepo(context = context)
+    fun providesDataStore(sharedPreferences: SharedPreferences) =
+        DataStoreRepo(sharedPreferences)
+
+    @Provides
+    @Singleton
+    fun provideEncryptedPreferences(@ApplicationContext context: Context): SharedPreferences {
+
+        // Step 1: Create or retrieve the Master Key for encryption/decryption
+        val masterKey = MasterKey.Builder(context, MasterKey.DEFAULT_MASTER_KEY_ALIAS)
```
Comment (pinakinkansara) [inline app/src/main/java/ca/bc/gov/bchealth/di/DataStoreModule.kt:35]: MasterKey.DEFAULT_MASTER_KEY_ALIAS) is optional, it used this by default

**Review by pinakinkansara — CHANGES_REQUESTED** [review]

File: app/src/main/java/ca/bc/gov/bchealth/datasource/DataStoreRepo.kt (hunk 1 above)
Comment (amit-metri) [reply app/src/main/java/ca/bc/gov/bchealth/datasource/DataStoreRepo.kt:68]: There is no suspend function to be executed while saving encrypted preferences(SharedPreferenses.editor.apply()). Therefore I kept this as a normal function.

File: app/src/main/java/ca/bc/gov/bchealth/di/DataStoreModule.kt (hunk 5 above)
Comment (amit-metri) [reply app/src/main/java/ca/bc/gov/bchealth/di/DataStoreModule.kt:35]: Alias is removed.

**Comment by amit-metri** [comment]
Closing this PR as shared preferences encryption is covered in sqlite encryption.

**Pull request closed without merging**